packages/components/src/mobile/color-settings/index.native.js

```diff
@@ -176,18 +193,45 @@ function ColorSettings( {
 						isGradientColor={ isGradientColor }
 						currentSegment={ currentSegment }
 						isCustomScreen={ isCustomScreen }
-						onCustomPress={ () => {
-							onCustomScreenToggle( true );
-						} }
+						onCustomPress={ onCustomPress }
 						shouldEnableBottomSheetScroll={
 							shouldEnableBottomSheetScroll
 						}
 						defaultSettings={ defaultSettings }
 					/>
+					{ isCustomGadientShown && (
+						<>
+							<View style={ horizontalSeparatorStyle } />
+							<PanelBody>
+								<ColorControl
+									label={ __( 'Customize Gradient' ) }
+									onPress={ () =>
+										onCustomGradientScreenToggle( true )
+									}
+									withColorIndicator={ false }
+								/>
+							</PanelBody>
+						</>
+					) }
 					<View style={ horizontalSeparatorStyle } />
 					{ getFooter() }
 				</View>
 			) }
+			{ isCustomGradientScreen && (
+				<View>
+					<NavigationHeader
+						screen={ __( 'Customize Gradient' ) }
+						leftButtonOnPress={ () =>
+							onCustomGradientScreenToggle( false )
+						}
+					/>
+					<CustomGradientPicker
+						setColor={ setColor }
+						currentValue={ currentValue }
+						isGradientColor={ isGradientColor }
+					/>
+				</View>
+			) }
 		</View>
```

Comment:
The ColorSettings component becomes quite big because of the `LayoutAnimations` limitations on Android. I'm happy that we could split it after adding navigation to the bottom-sheet. However, i see that i would need to add these changes to my implementation since i already refactored the color-settings in my PR but w/o the gradient customization.